app/src/main/java/helium314/keyboard/latin/settings/Settings.java

```diff
@@ -230,7 +234,7 @@ public void onDestroy() {
 
     @Override
     public void onSharedPreferenceChanged(final SharedPreferences prefs, final String key) {
-        if (dontReloadOnChanged.contains(key))
+        if (dontReloadOnChanged.contains(key) || key != null && key.startsWith(PREF_SAVED_APP_SUBTYPE_PREFIX))
```

Comment:
I think I commented similar cases before: please use parentheses to clarify order. While technically not necessary, it helps me with reading.

Reply:
Hmm, I don't really think I have anything close to guidelines for parentheses. For myself it's mostly if I find it a bit unclear I rather add parentheses. The `||` and `&&` order is one of the things that always confuses me for whatever reason, even though it's pretty straightforward.